src/Extensions/Microsoft.Diagnostics.Monitoring.Extension.Common/EgressHelper.cs

```diff
@@ -35,8 +38,27 @@ private static async Task<int> Egress<TOptions>(EgressProvider<TOptions> provide
             EgressArtifactResult result = new();
             try
             {
-                string jsonConfig = Console.ReadLine();
-                ExtensionEgressPayload configPayload = JsonSerializer.Deserialize<ExtensionEgressPayload>(jsonConfig);
+                StdInStream = Console.OpenStandardInput();
+
+                int dotnetMonitorPayloadProtocolVersion;
+                long payloadLengthBuffer;
+                byte[] payloadBuffer;
+
+                using (BinaryReader reader = new BinaryReader(StdInStream, Encoding.UTF8, leaveOpen: true))
+                {
+                    dotnetMonitorPayloadProtocolVersion = reader.ReadInt32();
+                    if (dotnetMonitorPayloadProtocolVersion != ExpectedPayloadProtocolVersion)
+                    {
+                        throw new EgressException(string.Format(CultureInfo.CurrentCulture, Strings.ErrorMessage_IncorrectPayloadVersion, dotnetMonitorPayloadProtocolVersion, ExpectedPayloadProtocolVersion));
+                    }
+
+                    payloadLengthBuffer = reader.ReadInt64();
+                }
+
+                payloadBuffer = new byte[payloadLengthBuffer];
```

Comment:
Can you validate that the length is not less than zero, as in, throw an ArgumentOutOfRangeException if it is less than zero?